combustion-android-ble/src/main/java/inc/combustion/framework/ble/DataLinkArbitrator.kt

```diff
@@ -121,15 +122,40 @@ internal class DataLinkArbitrator(
             } == null && !directInRange
         }
 
-    fun finish() {
-        networkNodes.forEach { it.value.finish() }
+    /**
+     * Cleans up all resources associated with this arbitrator, including disconnecting from a
+     * direct link to the probe (if available) and disconnecting from all MeatNet nodes in
+     * [deviceIdsToDisconnect].
+     *
+     * There's a couple specific things that come out of unlinking a probe that need to be addressed
+     * here:
+     *
+     * - Jobs are created on repeated probes (nodes) that need to be cancelled so that we don't
+     *   continue to obtain data for probes that we're disconnected from. If all jobs on a node are
+     *   blindly cancelled, then we'll likely cancel jobs that are still needed for other probes
+     *   connected to this node. The [jobKey] parameter allows for selective cancellation of jobs.
+     *   This is almost certainly the serial number.
+     * - On a related note, we need to be able to selectively disconnect from nodes as some are
+     *   still providing data from other probes. The [deviceIdsToDisconnect] parameter allows for
+     *   this selection.
+     */
+    fun finish(jobKey: String? = null, deviceIdsToDisconnect: Set<DeviceID>? = null) {
+        Log.d(LOG_TAG, "DataLinkArbitrator.finish($jobKey, $deviceIdsToDisconnect)")
+        networkNodes.values
+            .forEach {
+                it.finish(jobKey, deviceIdsToDisconnect?.contains(it.id) ?: true)
+            }
+
+        directLink?.disconnect()
```

Comment:
Disconnect here is a little bit hidden.  Not sure if it can be done higher up, but I'm not sure if we take explicit BLE actions otherwise in this class.

Reply:
Good call. Addressed in follow-up commit by moving this functionality up to the caller.